src/views/FismaTable/FismaTable.tsx

```diff
@@ -1,17 +1,123 @@
 import { FismaSystemType } from '@/types'
-import { DataGrid, GridColDef } from '@mui/x-data-grid'
-import { Box } from '@mui/material'
+import {
+  DataGrid,
+  GridColDef,
+  GridFooterContainer,
+  GridSlotsComponentsProps,
+  GridFooter,
+  GridRowId,
+} from '@mui/x-data-grid'
+import Tooltip from '@mui/material/Tooltip'
+import { Box, IconButton } from '@mui/material'
 import { useState } from 'react'
-
+import FileDownloadSharpIcon from '@mui/icons-material/FileDownloadSharp'
 import Link from '@mui/material/Link'
 import QuestionnareModal from '../QuestionnareModal/QuestionnareModal'
+import CustomSnackbar from '../Snackbar/Snackbar'
+import axiosInstance from '@/axiosConfig'
+
 type FismaTable2Props = {
   fismaSystems: FismaSystemType[]
   scores: Record<number, number>
 }
+
+type selectedRowsType = GridRowId[]
+declare module '@mui/x-data-grid' {
+  interface FooterPropsOverrides {
+    selectedRows: selectedRowsType
+    fismaSystems: FismaSystemType[]
+  }
+}
+export function CustomFooterSaveComponent(
+  props: NonNullable<GridSlotsComponentsProps['footer']>
+) {
+  const [openSnackbar, setOpenSnackbar] = useState<boolean>(false)
+  const handleCloseSnackbar = () => {
+    setOpenSnackbar(false)
+  }
+  const saveSystemAnswers = async () => {
+    if (props.selectedRows && props.selectedRows.length === 0) {
+      setOpenSnackbar(true)
+    } else {
+      let exportUrl = '/datacalls/2/export'
+      let exportFileName = ''
+      if (
+        props.selectedRows &&
+        props.fismaSystems &&
+        props.selectedRows.length < props.fismaSystems.length
+      ) {
+        exportUrl += '?'
+        exportFileName = 'Selected_Systems_Export.xlsx'
```

Comment:
The backend already provides the generated filename in the response headers. Remove this so the browser uses what the backend provides

Reply:
https://github.com/CMS-Enterprise/ztmf-ui/blob/c010edd8990bf658a21dbd486c9d0d321ddc0d54/src/views/FismaTable/FismaTable.tsx#L68-L69